### ISSUE TITLE
cns Failed

### ISSUE THREAD

**Issue by Gon1976** [marbl/canu#1964]
Hi,
I am running Canu 2.1.1 and I have an error similar to #1780.
The job

```
-- Running jobs.  Second attempt out of 2.
----------------------------------------
-- Starting 'cns' concurrent execution on Thu Jun 17 13:51:01 2021 with 1053.405 GB free disk space (1 processes; 6 concurrently)

    cd unitigging/5-consensus
    ./consensus.sh 14 > ./consensus.000014.out 2>&1

-- Finished on Thu Jun 17 13:51:02 2021 (one second) with 1053.405 GB free disk space
----------------------------------------
-- Using slow alignment for consensus (iteration '2').
-- Configured 18 consensus jobs.
--
-- Consensus jobs failed, tried 2 times, giving up.
--   job ctgcns/0014.cns FAILED.
--

ABORT:
ABORT: canu 2.1.1
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
```

I try possible solution reading the posts:
1. Set canu option cnsMaxCoverage=0, remove 5-consensus/consensus.sh and restart. Not Work
2. Run a test consensus to make sure it's the one that is failing:
```
utgcns -V -V -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 5 -O test.cns -maxcoverage 10 -e 0.2 -tig 5
-- Using seqFile '../AM64.ctgStore/partition.0014'.
-- Opening tigStore '../AM64.ctgStore' version 1.
-- Opening output results file 'test.cns'.
--
-- Computing consensus for b=5 to e=5 with errorRate 0.2000 (max 0.4000) and minimum overlap 500
--
                           ----------CONTAINED READS----------  -DOVETAIL  READS-
  tigID    length   reads      used coverage  ignored coverage      used coverage
------- --------- -------  -------- -------- -------- --------  -------- --------
Loading corrected-trimmed reads from seqFile '../AM64.ctgStore/partition.0014'

Processed 0 tigs and 0 singletons.

Consensus finished successfully.

Bye.
```
It's strange, because finish successfully but with 0 tigs. It's possible?
Here the error if I try the export utgncs result :
```
utgcns  -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 5  -tig 5 -export bad_tig.tig
-- Using seqFile '../AM64.ctgStore/partition.0014'.
-- Opening tigStore '../AM64.ctgStore' version 1.
-- Opening output package 'bad_tig.tig'.

Failed with 'Segmentation fault'; backtrace (libbacktrace):
utility/src/utility/system-stackTrace.C::83 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
stores/sqStore.C::136 in _ZN7sqStore24sqStore_saveReadToBufferEP11writeBufferjP6sqReadP16sqReadDataWriter()
stores/tgTig.C::701 in _ZN5tgTig10exportDataEP11writeBufferP7sqStoreb()
utgcns/utgcns.C::445 in _Z10exportTigsR13cnsParameters()
utgcns/utgcns.C::984 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Violación de segmento (`core' generado)
```
3. I try to change to -nounalign (no work)
$bin/utgcns -R ../AM64.ctgStore/partition.$jobid -T ../AM64.ctgStore 1 -P $jobid -O ./ctgcns/$j
obid.cns.WORKING -maxcoverage 40 -e 0.105 -nounalign -edlib -threads 8 

4. I look the files unitigConsensus.C and Consensus.pm and looks ok.

Thanks in advance,
Gonzalo

**Comment by skoren** [comment]
The output in the failed consensus file (```consensus.000014.out```) would have more details on the failed tig. I don't think it's tig 5, that shouldn't end up in partition 14 and the command for it is also incorrect since the partition (the `-P` value shouldn't be 5). I also don't think cnsMaxCoverage would help as that bug was fixed in the 2.1.1 release. The `-nounalign` isn't a valid option, I think you meant `-norealign`.

Try running:
```
$bin/utgcns -R ../AM64.ctgStore/partition.0014-T ../AM64.ctgStore 1 -P 14-O tmp.cns -maxcoverage 40 -e 0.105 -nounalign -edlib -threads 1 -V -V -V -norealign
```
and post the output assuming it fails.

**Comment by Gon1976** [comment]
Thanks @skoren, I run your suggestion, i have this error
[nanocruzi@hpc02 5-consensus]$ utgcns -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 14 -O tmp.cns -maxcoverage 40 -e 0.105 -nounalign -edlib -threads 1 -V -V -V -norealign

Unknown option '-nounalign'.

then, I run without -nounalign, and here the result:
[nanocruzi@hpc02 5-consensus]$ utgcns -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 14 -O tmp.cns -maxcoverage 40 -e 0.105 -edlib -threads 1 -V -V -V -norealign
-- Using seqFile '../AM64.ctgStore/partition.0014'.
-- Opening tigStore '../AM64.ctgStore' version 1.
-- Opening output results file 'tmp.cns'.
--
-- Computing consensus for b=0 to e=2176 with errorRate 0.1050 (max 0.4000) and minimum overlap 500
--
                           ----------CONTAINED READS----------  -DOVETAIL  READS-
  tigID    length   reads      used coverage  ignored coverage      used coverage
------- --------- -------  -------- -------- -------- --------  -------- --------
Loading corrected-trimmed reads from seqFile '../AM64.ctgStore/partition.0014'
      5    481540     355       324   10.44x        0    0.00x        31    3.24x

generateTemplateStitch()-- COPY READ read #0 3234 (len=14520 to 0-14526)

generateTemplateStitch()-- read #11/355 ident 71335 position 14182-41548 SAVE THIN FIRST
generateTemplateStitch()-- WARNING, increasing min overlap from 344 to 27306 for read 11 (14182 - 41548

generateTemplateStitch()-- ALIGN template -10056-14520 (len=24576) to read #11 71335 0-27306 (len=27306 actual=27306 at 14182-41548)  expecting olap of 27306

Failed with 'Segmentation fault'; backtrace (libbacktrace):
utility/src/utility/system-stackTrace.C::83 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
utility/src/utility/edlib.C::1589 in transformSequences()
utility/src/utility/edlib.C::198 in _Z10edlibAlignPKciS0_i16EdlibAlignConfig()
utgcns/unitigConsensus.C::424 in _ZN15unitigConsensus22generateTemplateStitchEv()
utgcns/unitigConsensus.C::787 in _ZN15unitigConsensus13generatePBDAGEP5tgTigcPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/unitigConsensus.C::1431 in _ZN15unitigConsensus8generateEP5tgTigccPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/utgcns.C::614 in _Z11processTigsR13cnsParameters()
utgcns/utgcns.C::990 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Violación de segmento (`core' generado)

**Comment by skoren** [comment]
It should be `-norealign` but that's not the issue so don't worry about adding it. This looks the same as #1901 (https://github.com/marbl/canu/commit/d491848187ce8a13b7654b4852a9453ffdbf89ef). You can try updating your code with those changes or pulling that commit and seeing if it fixes the error.

**Comment by Gon1976** [comment]
Ok I change unitigConsensus.C as the commit mention. I re-run utgcns but have the same error.
utgcns -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 14 -O tmp.cns -maxcoverage 40 -e 0.3 -edlib -threads 1 -V -V -V -V -V -norealign
-- Using seqFile '../AM64.ctgStore/partition.0014'.
-- Opening tigStore '../AM64.ctgStore' version 1.
-- Opening output results file 'tmp.cns'.
--
-- Computing consensus for b=0 to e=2176 with errorRate 0.3000 (max 0.4000) and minimum overlap 500
--
                           ----------CONTAINED READS----------  -DOVETAIL  READS-
  tigID    length   reads      used coverage  ignored coverage      used coverage
------- --------- -------  -------- -------- -------- --------  -------- --------
Loading corrected-trimmed reads from seqFile '../AM64.ctgStore/partition.0014'
      5    481540     355       324   10.44x        0    0.00x        31    3.24x

generateTemplateStitch()-- COPY READ read #0 3234 (len=14520 to 0-14526)

generateTemplateStitch()-- read #11/355 ident 71335 position 14182-41548 SAVE THIN FIRST
generateTemplateStitch()-- WARNING, increasing min overlap from 344 to 27306 for read 11 (14182 - 41548

generateTemplateStitch()-- ALIGN template -10056-14520 (len=24576) to read #11 71335 0-27306 (len=27306 actual=27306 at 14182-41548)  expecting olap of 27306

Failed with 'Segmentation fault'; backtrace (libbacktrace):
utility/src/utility/system-stackTrace.C::83 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
utility/src/utility/edlib.C::1589 in transformSequences()
utility/src/utility/edlib.C::198 in _Z10edlibAlignPKciS0_i16EdlibAlignConfig()
utgcns/unitigConsensus.C::424 in _ZN15unitigConsensus22generateTemplateStitchEv()
utgcns/unitigConsensus.C::787 in _ZN15unitigConsensus13generatePBDAGEP5tgTigcPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/unitigConsensus.C::1431 in _ZN15unitigConsensus8generateEP5tgTigccPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/utgcns.C::614 in _Z11processTigsR13cnsParameters()
utgcns/utgcns.C::990 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()


Or I should to delete consensus.sh and re-start Canu? Sorry but maybe I misunderstand the solution.
Gonzalo

**Comment by skoren** [comment]
Did you re-compile canu after making the changes to the .C file? You need to re-build the code and make sure to check that consensus.sh to points to the newly compiled utgcns binary not the old one.

**Comment by Gon1976** [comment]
thanks, not is the anser! I will try! thanks again
Gonzalo

**Comment by Gon1976** [comment]
Thanks Skoren! I will try it!

**Comment by Gon1976** [comment]
Hi Skoren, finally, today I can test the commit.
I think I succesefully compile with the 
Building release v2.1.1
For 'Linux' '4.18.0-147.5.1.el8_1.x86_64' as 'amd64' into '/home/nanocruzi/installs/canu-2.1.1/build/{bin,obj}'.
Using '/usr/bin/g++' version '8.3.1'.

g++ -o /home/nanocruzi/installs/canu-2.1.1/build/obj/bin/utgcns/utgcns/unitigConsensus.o -c -MD -g3 -O4 -funroll-loops -fexpensive-optimizations -finline-functions -fomit-frame-pointer -Wall -Wextra -Wformat -Wno-char-subscripts -Wno-sign-compare -Wno-unused-function -Wno-unused-parameter -Wno-unused-variable -Wno-deprecated-declarations -Wno-format-truncation -std=c++11 -pthread -fopenmp -fPIC -DLIBBACKTRACE -I/home/nanocruzi/installs/canu-2.1.1/src -Iutility/src/utility -Istores -IoverlapInCore/libedlib -Iutgcns/libpbutgcns -Iutgcns/libboost utgcns/unitigConsensus.C
g++ -o /home/nanocruzi/installs/canu-2.1.1/build/bin/utgcns -pthread -fopenmp -lm -L/home/nanocruzi/installs/canu-2.1.1/build/lib /home/nanocruzi/installs/canu-2.1.1/build/obj/bin/utgcns/utgcns/utgcns.o /home/nanocruzi/installs/canu-2.1.1/build/obj/bin/utgcns/utgcns/stashContains.o /home/nanocruzi/installs/canu-2.1.1/build/obj/bin/utgcns/utgcns/unitigConsensus.o -lcanu

Success!
canu installed in /home/nanocruzi/installs/canu-2.1.1/build/bin/canu

Then I run utgcns for partition failed (14). I obtain a different output but I have this error at the end:
utgcns -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 14 -O tmp.cns -maxcoverage 40 -e 0.3 -edlib -threads 1 -V -V -V -V -V -norealign
-- Using seqFile '../AM64.ctgStore/partition.0014'.
-- Opening tigStore '../AM64.ctgStore' version 1.
-- Opening output results file 'tmp.cns'.
--
-- Computing consensus for b=0 to e=2176 with errorRate 0.3000 (max 0.4000) and minimum overlap 500
--
                           ----------CONTAINED READS----------  -DOVETAIL  READS-
  tigID    length   reads      used coverage  ignored coverage      used coverage
------- --------- -------  -------- -------- -------- --------  -------- --------
Loading corrected-trimmed reads from seqFile '../AM64.ctgStore/partition.0014'
      5    481540     355       324   10.44x        0    0.00x        31    3.24x

generateTemplateStitch()-- COPY READ read #0 3234 (len=14520 to 0-14526)

generateTemplateStitch()-- read #11/355 ident 71335 position 14182-41548 SAVE THIN FIRST
generateTemplateStitch()-- WARNING, increasing min overlap from 344 to 500 for read 11 (14182 - 41548)

generateTemplateStitch()-- ALIGN template 14070-14520 (len=450) to read #11 71335 0-550 (len=550 actual=27306 at 14182-41548)  expecting olap of 500
generateTemplateStitch()-- FAILED to align - no result
generateTemplateStitch()-- FAILED to align - no result - decrease template size by 10%
generateTemplateStitch()-- FAILED to align - no result - increase read size by 10%

generateTemplateStitch()-- ALIGN template 14120-14520 (len=400) to read #11 71335 0-600 (len=600 actual=27306 at 14182-41548)  expecting olap of 500
generateTemplateStitch()-- FAILED to align - no result
generateTemplateStitch()-- FAILED to align - no result - decrease template size by 10%
generateTemplateStitch()-- FAILED to align - no result - increase read size by 10%

generateTemplateStitch()-- ALIGN template 14169-14520 (len=351) to read #11 71335 0-651 (len=651 actual=27306 at 14182-41548)  expecting olap of 500
generateTemplateStitch()-- FOUND alignment at 0-334 editDist 28 alignLen 355 8%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 14169-14520 to read 11 0-334; copy read 334-27306 to template.
generateTemplateStitch()-- Append read from 334 to 27306, starting at tiglen 14520
generateTemplateStitch()-- Template now length 41492, expected 41548, difference -0.1349%

generateTemplateStitch()-- read #13/355 ident 39045 position 14844-47679 SAVE FIRST
generateTemplateStitch()-- read #19/355 ident 28190 position 22302-69485 SAVE
generateTemplateStitch()-- read #20/355 ident 96827 position 22827-49266
generateTemplateStitch()-- read #22/355 ident 80904 position 24300-79165 SAVE
generateTemplateStitch()-- read #23/355 ident 83072 position 25779-61134
generateTemplateStitch()-- read #24/355 ident 13916 position 29956-67769
generateTemplateStitch()-- read #26/355 ident 16124 position 30862-46820
generateTemplateStitch()-- read #27/355 ident 72619 position 32990-64415
generateTemplateStitch()-- read #29/355 ident 33840 position 35652-41686
generateTemplateStitch()-- read #30/355 ident 87987 position 38932-56989
generateTemplateStitch()-- read #31/355 ident 60805 position 39766-59136
generateTemplateStitch()-- read #32/355 ident 93065 position 39805-48661
generateTemplateStitch()-- read #33/355 ident 107055 position 39862-48564
generateTemplateStitch()-- read #34/355 ident 105443 position 40057-51442
generateTemplateStitch()-- read #35/355 ident 86087 position 40447-44321
generateTemplateStitch()-- read #36/355 ident 76333 position 41327-44708 THIN

generateTemplateStitch()-- ALIGN template 25968-41492 (len=15524) to read #22 80904 0-18973 (len=18973 actual=54718 at 24300-79165)  expecting olap of 17248
generateTemplateStitch()-- FOUND alignment at 1656-17218 editDist 376 alignLen 15701 2%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 25968-41492 to read 22 1656-17218; copy read 17218-54718 to template.
generateTemplateStitch()-- Append read from 17218 to 54718, starting at tiglen 41492
generateTemplateStitch()-- Template now length 78992, expected 79165, difference -0.2188%

generateTemplateStitch()-- read #39/355 ident 74503 position 44413-119437 SAVE FIRST
generateTemplateStitch()-- read #42/355 ident 84804 position 45204-104023
generateTemplateStitch()-- read #56/355 ident 74595 position 53965-115360
generateTemplateStitch()-- read #58/355 ident 12951 position 56745-82590
generateTemplateStitch()-- read #60/355 ident 21946 position 62026-92289
generateTemplateStitch()-- read #61/355 ident 57923 position 64053-102053
generateTemplateStitch()-- read #67/355 ident 18578 position 69027-84847
generateTemplateStitch()-- read #69/355 ident 30009 position 70724-89765
generateTemplateStitch()-- read #70/355 ident 21478 position 72205-85695
generateTemplateStitch()-- read #71/355 ident 97897 position 77454-116805
generateTemplateStitch()-- read #72/355 ident 94112 position 78217-112680
generateTemplateStitch()-- read #73/355 ident 101611 position 78335-82825
generateTemplateStitch()-- read #74/355 ident 84791 position 78663-112603
generateTemplateStitch()-- read #75/355 ident 67907 position 80184-85850 THIN

generateTemplateStitch()-- ALIGN template 47715-78992 (len=31277) to read #39 74503 0-38228 (len=38228 actual=74771 at 44413-119437)  expecting olap of 34752
generateTemplateStitch()-- FOUND alignment at 3336-34611 editDist 1386 alignLen 31863 4%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 47715-78992 to read 39 3336-34611; copy read 34611-74771 to template.
generateTemplateStitch()-- Append read from 34611 to 74771, starting at tiglen 78992
generateTemplateStitch()-- Template now length 119152, expected 119437, difference -0.2389%

generateTemplateStitch()-- read #79/355 ident 61132 position 82102-136618 SAVE FIRST
generateTemplateStitch()-- read #83/355 ident 30133 position 92587-133740
generateTemplateStitch()-- read #85/355 ident 21932 position 93227-138888 SAVE
generateTemplateStitch()-- read #86/355 ident 10132 position 93542-155359 SAVE
generateTemplateStitch()-- read #87/355 ident 109660 position 94291-142854
generateTemplateStitch()-- read #90/355 ident 61465 position 98318-172022 SAVE
generateTemplateStitch()-- read #92/355 ident 71885 position 101292-124479
generateTemplateStitch()-- read #93/355 ident 86694 position 101663-145665
generateTemplateStitch()-- read #94/355 ident 84761 position 101871-163119
generateTemplateStitch()-- read #95/355 ident 100315 position 104297-129879
generateTemplateStitch()-- read #100/355 ident 65192 position 110219-124733
generateTemplateStitch()-- read #102/355 ident 34530 position 113096-135102
generateTemplateStitch()-- read #103/355 ident 100004 position 117047-120908
generateTemplateStitch()-- read #104/355 ident 92227 position 117584-127059
generateTemplateStitch()-- read #105/355 ident 91445 position 118071-122545
generateTemplateStitch()-- read #106/355 ident 10604 position 123057-143761 THIN

generateTemplateStitch()-- ALIGN template 100144-119152 (len=19008) to read #90 61465 0-23231 (len=23231 actual=73395 at 98318-172022)  expecting olap of 21119
generateTemplateStitch()-- FOUND alignment at 2061-20833 editDist 1104 alignLen 19359 6%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 100144-119152 to read 90 2061-20833; copy read 20833-73395 to template.
generateTemplateStitch()-- Append read from 20833 to 73395, starting at tiglen 119152
generateTemplateStitch()-- Template now length 171714, expected 172022, difference -0.1792%

generateTemplateStitch()-- read #130/355 ident 67990 position 145180-190309 SAVE FIRST
generateTemplateStitch()-- read #133/355 ident 740 position 147545-197708 SAVE
generateTemplateStitch()-- read #136/355 ident 17033 position 149272-173030
generateTemplateStitch()-- read #137/355 ident 96331 position 151228-175025
generateTemplateStitch()-- read #141/355 ident 86417 position 156305-189783
generateTemplateStitch()-- read #142/355 ident 104450 position 157214-189781
generateTemplateStitch()-- read #143/355 ident 21864 position 157677-181482
generateTemplateStitch()-- read #145/355 ident 53980 position 159805-174958
generateTemplateStitch()-- read #148/355 ident 87860 position 165710-176551
generateTemplateStitch()-- read #149/355 ident 107711 position 168877-174674
generateTemplateStitch()-- read #150/355 ident 81548 position 171138-174221
generateTemplateStitch()-- read #151/355 ident 15522 position 171320-248576 SAVE
generateTemplateStitch()-- read #152/355 ident 15214 position 173332-230821 THIN

generateTemplateStitch()-- ALIGN template 171082-171714 (len=632) to read #151 15522 0-773 (len=773 actual=76731 at 171320-248576)  expecting olap of 702
generateTemplateStitch()-- FAILED to align - no result
generateTemplateStitch()-- FAILED to align - no result - decrease template size by 10%
generateTemplateStitch()-- FAILED to align - no result - increase read size by 10%

generateTemplateStitch()-- ALIGN template 171152-171714 (len=562) to read #151 15522 0-843 (len=843 actual=76731 at 171320-248576)  expecting olap of 702
generateTemplateStitch()-- FAILED to align - no result
generateTemplateStitch()-- FAILED to align - no result - decrease template size by 10%
generateTemplateStitch()-- FAILED to align - no result - increase read size by 10%

generateTemplateStitch()-- ALIGN template 171222-171714 (len=492) to read #151 15522 0-913 (len=913 actual=76731 at 171320-248576)  expecting olap of 702
generateTemplateStitch()-- FOUND alignment at 357-873 editDist 40 alignLen 522 8%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 171222-171714 to read 151 357-873; copy read 873-76731 to template.
generateTemplateStitch()-- Append read from 873 to 76731, starting at tiglen 171714
generateTemplateStitch()-- Template now length 247572, expected 248576, difference -0.4047%

generateTemplateStitch()-- read #167/355 ident 78763 position 203287-261825 SAVE FIRST
generateTemplateStitch()-- read #171/355 ident 12286 position 208400-261093
generateTemplateStitch()-- read #176/355 ident 4544 position 213734-268476 SAVE
generateTemplateStitch()-- read #181/355 ident 26992 position 221808-260285
generateTemplateStitch()-- read #182/355 ident 17313 position 224367-250509
generateTemplateStitch()-- read #184/355 ident 297 position 227395-249970
generateTemplateStitch()-- read #189/355 ident 93570 position 230892-254912
generateTemplateStitch()-- read #192/355 ident 107647 position 235472-263503
generateTemplateStitch()-- read #194/355 ident 53514 position 235904-268249
generateTemplateStitch()-- read #195/355 ident 20854 position 236184-253238
generateTemplateStitch()-- read #198/355 ident 101212 position 237308-254781
generateTemplateStitch()-- read #200/355 ident 84601 position 239418-306327 SAVE
generateTemplateStitch()-- read #201/355 ident 69840 position 239567-286405
generateTemplateStitch()-- read #202/355 ident 51569 position 239761-290207
generateTemplateStitch()-- read #204/355 ident 66316 position 242058-265040
generateTemplateStitch()-- read #209/355 ident 65621 position 249093-321115 THIN

generateTemplateStitch()-- ALIGN template 239329-247572 (len=8243) to read #200 84601 0-10074 (len=10074 actual=66661 at 239418-306327)  expecting olap of 9158
generateTemplateStitch()-- FOUND alignment at 795-9172 editDist 482 alignLen 8499 6%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 239329-247572 to read 200 795-9172; copy read 9172-66661 to template.
generateTemplateStitch()-- Append read from 9172 to 66661, starting at tiglen 247572
generateTemplateStitch()-- Template now length 305061, expected 306327, difference -0.4141%

generateTemplateStitch()-- read #209/355 ident 65621 position 249093-321115 SAVE FIRST
generateTemplateStitch()-- read #215/355 ident 90486 position 258959-310832
generateTemplateStitch()-- read #222/355 ident 84800 position 268319-319082
generateTemplateStitch()-- read #225/355 ident 81152 position 270886-329186 SAVE
generateTemplateStitch()-- read #227/355 ident 33315 position 272354-323423
generateTemplateStitch()-- read #230/355 ident 90468 position 273485-326807
generateTemplateStitch()-- read #231/355 ident 74479 position 278775-320499
generateTemplateStitch()-- read #233/355 ident 78899 position 279355-320914
generateTemplateStitch()-- read #241/355 ident 46454 position 290107-317098
generateTemplateStitch()-- read #242/355 ident 34146 position 290972-329972 SAVE
generateTemplateStitch()-- read #244/355 ident 90127 position 294172-318582
generateTemplateStitch()-- read #246/355 ident 99735 position 295688-324452
generateTemplateStitch()-- read #247/355 ident 45646 position 295773-327340
generateTemplateStitch()-- read #248/355 ident 90588 position 296664-335012 SAVE
generateTemplateStitch()-- read #252/355 ident 28034 position 309456-341343 THIN

generateTemplateStitch()-- ALIGN template 296364-305061 (len=8697) to read #248 90588 0-10630 (len=10630 actual=38241 at 296664-335012)  expecting olap of 9663
generateTemplateStitch()-- FOUND alignment at 728-9624 editDist 528 alignLen 9019 6%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 296364-305061 to read 248 728-9624; copy read 9624-38241 to template.
generateTemplateStitch()-- Append read from 9624 to 38241, starting at tiglen 305061
generateTemplateStitch()-- Template now length 333678, expected 335012, difference -0.3990%

generateTemplateStitch()-- read #252/355 ident 28034 position 309456-341343 SAVE FIRST
generateTemplateStitch()-- read #253/355 ident 24239 position 312804-364546 SAVE
generateTemplateStitch()-- read #256/355 ident 61119 position 312925-336782
generateTemplateStitch()-- read #257/355 ident 42959 position 316486-353778
generateTemplateStitch()-- read #258/355 ident 91534 position 319642-355249
generateTemplateStitch()-- read #261/355 ident 23858 position 322270-364615 SAVE
generateTemplateStitch()-- read #262/355 ident 29384 position 325839-362605
generateTemplateStitch()-- read #263/355 ident 44577 position 329014-337036
generateTemplateStitch()-- read #264/355 ident 62993 position 329291-335749
generateTemplateStitch()-- read #265/355 ident 12136 position 329672-340468
generateTemplateStitch()-- read #266/355 ident 22914 position 330487-354972
generateTemplateStitch()-- read #267/355 ident 13427 position 339314-368447 THIN

generateTemplateStitch()-- ALIGN template 322210-333678 (len=11468) to read #261 23858 0-14017 (len=14017 actual=42283 at 322270-364615)  expecting olap of 12742
generateTemplateStitch()-- FOUND alignment at 1028-12601 editDist 781 alignLen 11781 7%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 322210-333678 to read 261 1028-12601; copy read 12601-42283 to template.
generateTemplateStitch()-- Append read from 12601 to 42283, starting at tiglen 333678
generateTemplateStitch()-- Template now length 363360, expected 364615, difference -0.3448%

generateTemplateStitch()-- read #267/355 ident 13427 position 339314-368447 SAVE FIRST
generateTemplateStitch()-- read #270/355 ident 66323 position 342719-374628 SAVE
generateTemplateStitch()-- read #273/355 ident 44922 position 345364-374008
generateTemplateStitch()-- read #275/355 ident 96329 position 345630-373430
generateTemplateStitch()-- read #279/355 ident 51964 position 347510-399552 SAVE
generateTemplateStitch()-- read #282/355 ident 64943 position 350696-395987
generateTemplateStitch()-- read #284/355 ident 6514 position 355371-379538
generateTemplateStitch()-- read #285/355 ident 74658 position 356460-371214
generateTemplateStitch()-- read #287/355 ident 17664 position 357268-397954
generateTemplateStitch()-- read #288/355 ident 6135 position 366830-370259 THIN

generateTemplateStitch()-- ALIGN template 347965-363360 (len=15395) to read #279 51964 0-18816 (len=18816 actual=51624 at 347510-399552)  expecting olap of 17105
generateTemplateStitch()-- FOUND alignment at 1841-17182 editDist 992 alignLen 15738 6%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 347965-363360 to read 279 1841-17182; copy read 17182-51624 to template.
generateTemplateStitch()-- Append read from 17182 to 51624, starting at tiglen 363360
generateTemplateStitch()-- Template now length 397802, expected 399552, difference -0.4390%

generateTemplateStitch()-- read #290/355 ident 67339 position 381171-412830 SAVE FIRST
generateTemplateStitch()-- read #291/355 ident 20421 position 381471-412485
generateTemplateStitch()-- read #292/355 ident 93423 position 381832-413988 SAVE
generateTemplateStitch()-- read #298/355 ident 42087 position 384797-433867 SAVE
generateTemplateStitch()-- read #300/355 ident 732 position 388117-460903 SAVE
generateTemplateStitch()-- read #301/355 ident 45647 position 388139-446494
generateTemplateStitch()-- read #305/355 ident 20858 position 392900-414054
generateTemplateStitch()-- read #307/355 ident 12988 position 392957-481540 SAVE
generateTemplateStitch()-- read #308/355 ident 78103 position 393304-400381
generateTemplateStitch()-- read #309/355 ident 43398 position 394775-400407
generateTemplateStitch()-- read #310/355 ident 20005 position 394889-414068
generateTemplateStitch()-- read #311/355 ident 107278 position 395548-411508
generateTemplateStitch()-- read #312/355 ident 102112 position 396528-400827
generateTemplateStitch()-- read #313/355 ident 87255 position 397394-434398
generateTemplateStitch()-- read #314/355 ident 10768 position 397559-442122
generateTemplateStitch()-- read #315/355 ident 36630 position 399070-403254 THIN

generateTemplateStitch()-- ALIGN template 391866-397802 (len=5936) to read #307 12988 0-7255 (len=7255 actual=87636 at 392957-481540)  expecting olap of 6595
generateTemplateStitch()-- FOUND alignment at 649-6558 editDist 313 alignLen 6025 5%
generateTemplateStitch()--
generateTemplateStitch()-- Aligned template 391866-397802 to read 307 649-6558; copy read 6558-87636 to template.
generateTemplateStitch()-- Append read from 6558 to 87636, starting at tiglen 397802
generateTemplateStitch()-- Template now length 478880, expected 481540, difference -0.5539%

generateTemplateStitch()-- NO MORE READS TO ALIGN

generateTemplateStitch()-- generated template of length 478880, expected length 481540, -0.5539% difference.
Generated template of length 478880
Aligning reads.
alignEdLib()-- align read    3234 eRate 0.1500 at         0-16623
Failed with 'Segmentation fault'; backtrace (libbacktrace):
utility/src/utility/system-stackTrace.C::83 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
(null)::0 in (null)()
utgcns/unitigConsensus.C::637 in _Z10alignEdLibR12dagAlignmentR10tgPositionPcjS3_jddb()
utgcns/unitigConsensus.C::814 in _ZN15unitigConsensus13generatePBDAGEP5tgTigcPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE._omp_fn.0()
(null)::0 in (null)()
utgcns/unitigConsensus.C::807 in _ZN15unitigConsensus13generatePBDAGEP5tgTigcPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/unitigConsensus.C::1436 in _ZN15unitigConsensus8generateEP5tgTigccPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEE()
utgcns/utgcns.C::614 in _Z11processTigsR13cnsParameters()
utgcns/utgcns.C::990 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Violación de segmento (`core' generado)

**Comment by skoren** [comment]
I think this may be a compilation issue, given that the alignment is failing on the very first read which is essentially being aligned back to itself. The trace also doesn't indicate anything I'm slightly suspicious of read 71335 because it caused the initial issue and does add some sequence to the start of the contig. 

You can get past the  error and finish the assembly by running utgcns with 
```
utgcns -R ../AM64.ctgStore/partition.0014 -T ../AM64.ctgStore 1 -P 14 -O tmp.cns -maxcoverage 40 -e 0.3 -quick -edlib -threads 1 -V -V -V -V -V
```
this won't generate as high a quality of a consensus as normal but should be OK. If you can share the data (we'd need the ctgStore folder, see the FAQ for instructions to send it to us), I can see if I can reproduce the latter error locally or not.

**Comment by Gon1976** [comment]
That's work! Thanks
I upload the content of ctgStore according to instructions
Now I should re-start canu to finish the assembly?
Gonzalo
Processed 9 tigs and 1 singleton.

Consensus finished successfully.

Bye.
I will shar

**Comment by skoren** [comment]
Yes, move the tmp.cns file to the correct location, something like`./ctgcns/00014.cns` so it matches the output from the other partitions. Then, you can re-start Canu.

**Comment by Gon1976** [comment]
great, assembly finished!
thanks

**Comment by skoren** [comment]
I was able to download the data you shared and ran it w/o issue with the bug fix from #1901 so I think it was a compilation problem.
